src/argilla/client/users.py

```diff
@@ -108,14 +109,16 @@ def __init__(
         raise Exception(error_msg)
 
     @property
-    @allowed_for_roles(roles=[UserRole.owner])
-    def workspaces(self) -> Optional[List[WorkspaceModel]]:
+    def workspaces(self) -> Optional[List["WorkspaceModel"]]:
         """Returns the workspace names the current user is linked to.
 
         Returns:
             A list of `WorkspaceModel` the current user is linked to.
         """
-        return users_api_v1.list_user_workspaces(self.__client, self.id).parsed
+        if self.role == UserRole.owner:
+            return users_api_v1.list_user_workspaces(self.__client, self.id).parsed
+        else:
+            return workspaces_api_v1.list_workspaces_me(self.__client).parsed
```

Comment:
this is working because the admin and annotator role cannot create `User` without using `User.me`, right?

Reply:
I mean, ideally, to launch the `list_workspaces_me` the condition should verify that `self` is the connected user. Anyway, this should work

Reply:
No, you're right, indeed if we list users from the `owner` we won't get the real results, I'll add the corrected statement now! Nice catch 😄

Reply:
This change will have a problem if the client is reinitialized after creating the user instance. We should find another approach